flytekit/image_spec/image_spec.py

```diff
@@ -114,6 +114,10 @@ def __post_init__(self):
         if self.source_root is not None:
             self.source_copy_mode = self.source_copy_mode or CopyFileDetection.LOADED_MODULES
 
+        builder_registry = ImageBuildEngine.get_registry()
+        if self.builder is None and builder_registry:
+            self.builder = max(builder_registry, key=lambda name: builder_registry[name][1])
```

Comment:
Can you leave a comment explaining the use of `max` here?

Reply:
added it